user/views.py

```diff
@@ -2,13 +2,62 @@
 from django.contrib.auth import login, logout, authenticate
 from django.contrib import messages
 from django.contrib.auth.forms import UserCreationForm, AuthenticationForm
-from django.http import HttpResponse, HttpResponseRedirect
+from django.http import HttpResponse, HttpResponseRedirect, JsonResponse
 from .forms import SignUpForm
 from user.models import Users
 from django.template import RequestContext
 from blog.models import Articles
-# Create your views here.
+from rest_framework import serializers, viewsets, status
+from rest_framework.parsers import JSONParser
+from django.views.decorators.csrf import csrf_exempt
+from django.contrib.auth import get_user_model
+from .serializers import UserSerializer, UserProfileSerializer
+import logging
+logger = logging.getLogger('django')
 
+User = get_user_model()
+
+@csrf_exempt
+def login_user(request):
+    if request.method == "POST":
+        if request.user.is_authenticated:
+            return HttpResponse(status = 400)
+        data = JSONParser().parse(request)
+        username = data["username"]
+        user = authenticate(username = username,password = data["password"])
+        if user:
+                login(request,user)
+                logger.info("User with username {} has been successfully logged in".format(username))
+                return(HttpResponse(status = 200))
+        else:
+            logger.info("User with username {} could not be logged in".format(username))
+            return(JsonResponse({"Error":"User with username {} could not be logged in".format(username)},status= 401))
+
+@csrf_exempt
+def logout_user(request):
+    if request.user.is_authenticated:
+        logout(request)
+        return(HttpResponse(status = 200))
+    else:
+        return(HttpResponse(status = 400))
+
+
+@csrf_exempt
+def register_user(request):
+    if request.method == "POST":
+        data = JSONParser().parse(request)
+        serializer = UserSerializer(data = data)
+        if serializer.is_valid():
+            serializer.save()
```

Comment:
What will happen if the user is saved and the user profile runs into an error. Shouldn't these statement wrapped in some kind to transaction block